apps/notifications/src/panel/templates/note.jsx

```diff
@@ -1,5 +1,5 @@
 import classNames from 'classnames';
-import React from 'react';
+import * as React from 'react';
```

Comment:
I wonder why this happened. 🤔

Reply:
`* as React` is actually the "right" way to import React, because it lets the bunder tree shake and remove the React exports that are not used. At this moment it's just theory that doesn't have any observable effect, but in the future `* as React` could lead to real optimizations.

Reply:
TIL

Reply:
Out of curiosity, is that true for any library?

Reply:
This threw me off because in other similar instances, the codemod refactored it to named import. So I was expecting to see:

```javascript
import { forwardRef } from 'react';
```

which should, in theory, be equivalent or better for tree-shaking purposes?

Reply:
maybe in this case, the file references the `React` object directly instead of just one import?

Reply:
> Out of curiosity, is that true for any library?

I guess it's true for any CJS library that uses `module.exports`:
```js
const React = {
  Component,
  useState,
};
module.exports = React;
```
Then when importing as ESM all of the following works:
```js
import React from 'react';
class Foo extends React.Component;
```
or
```js
import { Component } from 'react';
class Foo extends Component;
```
or
```js
import * as React from 'react';
class Foo extends React.Component;
```
All of them will always bundle the `useState` function, too, because neither the bundler not the minifier knows that `useState` is unused.

Now suppose some future version of React gets converted to ESM:
```js
export Component;
export useState;
```
Then `import React from 'react'` will stop working because the module doesn't have any default export. For the two remaining usages, bundler will be able to figure out that the `useState` export was never used and will optimize it out.

To make `import React` work in a back-compatible way, they'll need to add a default export:
```js
export default {
  Component,
  useState,
}
```
which, again, if it's never used, the bundler will be able to remove.